### ISSUE TITLE
ACCOUNT_JSON_INFO = 'https://www.instagram.com/{username}/?__a=1&__d=dis' no longer working

### ISSUE THREAD

**Issue by andre-santos1** [raiym/instagram-php-scraper#1134]
Hi there! It looks like that  ACCOUNT_JSON_INFO = 'https://www.instagram.com/{username}/?__a=1&__d=dis' is not working anymore.

Response is: Sorry, this page isn't available.
The link you followed may be broken, or the page may have been removed. [Go back to Instagram.](https://www.instagram.com/)

Does anyone know what I should add to the URL  'https://www.instagram.com/{username}' to get account json info?

Tks!

**Comment by beshoo** [comment]
I believe we're no longer in a golden age. We're now in a "shit age" due to Instagram's new rules. LOL. This JSON API will end soon!

**Comment by bjmeo8** [comment]
> Hi there! It looks like that ACCOUNT_JSON_INFO = '[https://www.instagram.com/{username}/?__a=1&__d=dis](https://www.instagram.com/%7Busername%7D/?__a=1&__d=dis)' is not working anymore.
> 
> Response is: Sorry, this page isn't available. The link you followed may be broken, or the page may have been removed. [Go back to Instagram.](https://www.instagram.com/)
> 
> Does anyone know what I should add to the URL '[https://www.instagram.com/{username}](https://www.instagram.com/%7Busername%7D)' to get account json info?
> 
> Tks!

Hello, I also have the same issue. I think they have probably changed the magic URL parameters "?__a=1&__d=dis".

I searched all over the internet and eventually found a solution on 
Reddit: (https://www.reddit.com/r/webscraping/comments/19fgptv/comment/kjjy2v7/?utm_source=share&utm_medium=web3x&utm_name=web3xcss&utm_term=1&utm_content=share_button) 

and on this web scraping service's blog: (https://scrapfly.io/blog/how-to-scrape-instagram/)[https://scrapfly.io/blog/how-to-scrape-instagram/) 

and their github (https://github.com/scrapfly/scrapfly-scrapers/blob/main/instagram-scraper/instagram.py](https://scrapfly.io/blog/how-to-scrape-instagram/)

To retrieve account information, you should proceed as follows:

`curl -X GET "https://i.instagram.com/api/v1/users/web_profile_info/?username=google" \
 -H "User-Agent: iphone_ua" \
 -H "x-ig-app-id: 936619743392459"
`

You can try the curl request directly here [https://reqbin.com/curl](https://reqbin.com/curl) and you will see that the response is a JSON containing the information accessible via the old method "?__a=1&__d=dis"

**Comment by beshoo** [comment]
![image](https://github.com/postaddictme/instagram-php-scraper/assets/4635115/371253eb-abb0-4b67-91b5-b95bc9e861f4)

**Comment by bjmeo8** [comment]
It seems there is a formatting issue with the initial request I submitted. 
Try this:

curl -X GET "https://i.instagram.com/api/v1/users/web_profile_info/?username=google" \
 -H "User-Agent: iphone_ua" \
 -H "x-ig-app-id: 936619743392459"
 
I just tried again, and it's working fine. Here's a screenshot:
 
![Screenshot 2024-01-28 at 11-16-25 Run Curl Commands Online](https://github.com/postaddictme/instagram-php-scraper/assets/68184289/5dd81ba7-fe04-4aa9-8f49-cb7593725173)

**Comment by beshoo** [comment]
It will end soon.

**Comment by bjmeo8** [comment]
You think? Why?

Perhaps the recent restriction of their public API is related to this?:
https://www.mcafee.com/blogs/internet-security/26-billion-records-released-the-mother-of-all-breaches/

Alternatively, they may be concerned that Instagram data is being massively scraped by competitors to train their AI models. Perhaps...

**Comment by beshoo** [comment]
I have been using an Instagram scraper since 2015, which enabled me to earn $10,000 each month. 
However, Instagram has recently started using AI to crack down on such activities. Although am were able to bypass the restrictions using residential proxies, Instagram is now taking further measures to put an end to this practice. 
They started with hashtags and are now targeting users. It seems that even the workarounds will soon be eliminated.

**Comment by bjmeo8** [comment]
Okay, I understand your point of view. But does that mean that all the bots, scraping services, etc., in the market will soon go extinct too? Wow, that will be a tough blow for many.

**Comment by beshoo** [comment]
I agree with you. And yes

**Comment by kvrushifa** [comment]
@beshoo how were you able to earn that much money just by scraping instagram ? :o

**Comment by beshoo** [comment]
"I didn't just scrape Instagram, I actually created a website called **instaranker.com**. It's a service where you can add your Instagram account and it will automatically engage with other users by liking, following, and saving their posts. 

I was able to make around $300 to $400 a day from 2016 to 2020. However, things started to change after 2021 website closed at the end of 2021..

I have another website called **viralcaption.com** that only scrapes data. I had to remove two functions which were user information and tag information. but am sure I will close the service very soon.

**Comment by andre-santos1** [comment]
I wasn't able to reproduce your steps. But it worked as an insight for me and I did as follows:

1. Changed my browser user-agent (try this browser extension for example: https://chromewebstore.google.com/detail/user-agent-switcher-for-c/djflhoibgkdhkhhcedjiklpkjnoahfmg) to this:


Mozilla/5.0 (Linux; Android 9; SM-A102U Build/PPR1.180610.011; wv) AppleWebKit/537.36 (KHTML, like Gecko) Version/4.0 Chrome/74.0.3729.136 Mobile Safari/537.36 Instagram 155.0.0.37.107 Android (28/9; 320dpi; 720x1468; samsung; SM-A102U; a10e; exynos7885; en_US; 239490550)

2. I pasted (without quotes) "https://i.instagram.com/api/v1/users/web_profile_info/?username={username}" on Chrome and I was able to get the JSON response. ({username} must be replaced by your target username)

3. To better visualize the results I recommend installing  JSON Beautifier & Editor extension on Chrome (https://chromewebstore.google.com/detail/json-beautifier-editor/lpopeocbeepakdnipejhlpcmifheolpl)

**Comment by jfrenl** [comment]
Yeah i found something called ?__user=0 it's probably useless information but maybe someone else can use this. Also ?__d=1 has a same kind of response.

**Comment by beshoo** [comment]
`__d` has to be `dis` only. 
ps: `__user=0` did nothing!

**Comment by jfrenl** [comment]
A bit ago have i always used __d=1 to get data too until it got patched and then i moved to __d=dis. I am not sure why it has been patched.

**Comment by beshoo** [comment]
> I have been using an Instagram scraper since 2015, which enabled me to earn $10,000 each month. However, Instagram has recently started using AI to crack down on such activities. Although am were able to bypass the restrictions using residential proxies, Instagram is now taking further measures to put an end to this practice. They started with hashtags and are now targeting users. It seems that even the workarounds will soon be eliminated.

@jfrenl

**Comment by jfrenl** [comment]
Yeah i read it, but still unfortunate that this is happening. I've been looking for solutions for a 2 weeks now. I hope to have a break trough soon.

**Comment by jfrenl** [comment]
I heard you were talking about proxies and i was wondering if you had any cheap providers for proxies. Because it doesn't matter if it is residential proxies or datacenter ips.

**Comment by jfrenl** [comment]
@beshoo I am not sure if you do know something about this but somehow this instagram url https://www.instagram.com/nike/?__a=1&__d=dis response back as status code 201. But normally you would get a 404 page from instagram.

**Comment by beshoo** [comment]
Even though, there is no json response....

**Comment by jfrenl** [comment]
Yeah that's right but any idea why statuscode 201 created?

**Comment by beshoo** [comment]
I don't know, but as I said, its helpless...

**Comment by megalancast** [comment]
I am wondering how paid APIs like https://rapidapi.com/neotank/api/simple-instagram-api/ work? I've checked it for a week and it seems to work pretty good, and `/account-info` endpoint returns exactly what `https://i.instagram.com/api/v1/users/web_profile_info/` endpoint returns (I think)...

**Comment by jfrenl** [comment]
> I am wondering how paid APIs like https://rapidapi.com/neotank/api/instagram130 work? I've checked it for a week and it seems to work pretty good, and `/account-info` endpoint returns exactly what `https://i.instagram.com/api/v1/users/web_profile_info/` endpoint returns (I think)...

I am pretty sure the person uses lots of accounts in the backend and due to people paying every month he can keep this project up. But I have a different solution that doesn't cost me anything.

**Comment by DerLomo** [comment]
> > I am wondering how paid APIs like https://rapidapi.com/neotank/api/instagram130 work? I've checked it for a week and it seems to work pretty good, and `/account-info` endpoint returns exactly what `https://i.instagram.com/api/v1/users/web_profile_info/` endpoint returns (I think)...
> 
> I am pretty sure the person uses lots of accounts in the backend and due to people paying every month he can keep this project up. But I have a different solution that doesn't cost me anything.

Can you share it?
I been looking for a bypass for ages now

**Comment by remontss** [comment]
https://i.instagram.com/api/v1/users/web_profile_info/?username=google 
change i.instagram.com to www
https://www.instagram.com/api/v1/users/web_profile_info/?username=google

**Comment by sshlyljt** [comment]
> https://i.instagram.com/api/v1/users/web_profile_info/?username=google 將 i.instagram.com 改為 www https://www.instagram.com/api/v1/users/web_profile_info/?username=google

{"message":"Please wait a few minutes before you try again.","require_login":true,"igweb_rollout":true,"status":"fail"}

The result has not changed

**Comment by jfrenl** [comment]
Yeah i think they finally patched this endpoint and maybe it still works for accounts but back when i tried it didn't work either.

**Comment by mjohnsoncode** [comment]
![Image](https://github.com/user-attachments/assets/a8799ab4-ad1b-4680-8225-4d8d358ec490)

All is still works without auth for /web_profile_info. You just need a proper TLS fingerprint and mobile proxy with IP rotation or multiport proxy. 

Need help? Feel free to [contact me](https://t.me/mjohnsoncode).
